packages/test/stochastic-test-utils/src/performActions.ts

```diff
@@ -219,20 +226,27 @@ export function performFuzzActions<
 
 	for (let operation = generator(state); operation !== done; operation = generator(state)) {
 		operations.push(operation);
-		if (saveInfo !== undefined && operations.length === saveInfo.saveAt) {
-			writeFileSync(saveInfo.filepath, JSON.stringify(operations));
+		if (saveInfo?.saveWhenFinish === false && operations.length === saveInfo.saveAt) {
```

Comment:
`saveInfo?.saveWhenFinish === false` will evaluate to `false` when `saveWhenFinish` is undefined.
This will prevent `saveAt` from working unless `saveWhenFinish` is explicitly set to `false`.